django/db/models/query.py

```diff
@@ -870,18 +865,14 @@ def annotate(self, *args, **kwargs):
         """
         annotations = OrderedDict()  # To preserve ordering of args
         for arg in args:
-            # The default_alias property may raise a TypeError, so we use
-            # a try/except construct rather than hasattr in order to remain
-            # consistent between PY2 and PY3 (hasattr would swallow
-            # the TypeError on PY2).
             try:
+                annotations[arg.default_alias] = arg
                 if arg.default_alias in kwargs:
```

Comment:
`TypeError` can be raised be `default_alias`, so I'd rather keep it there. At least, moving it outside will require checking for `TypeError` in a different way.